src/NzbDrone.Core/RootFolders/RootFolderService.cs

```diff
@@ -153,7 +154,34 @@ private List<UnmappedFolder> GetUnmappedFolders(string path)
             foreach (string unmappedFolder in unmappedFolders)
             {
                 var di = new DirectoryInfo(unmappedFolder.Normalize());
-                results.Add(new UnmappedFolder { Name = di.Name, Path = di.FullName });
+                var fileNfoName = String.Format("{0}\\tvshow.nfo", di.FullName);
+                if (File.Exists(fileNfoName))
+                {
+                    try
```

Comment:
The logic of pulling the ID from an NFO file should be pulled out to a separate method, possibly another class/interface so additional metadata types could be also be handled. It'd be worth looking at extending the existing metadata handling to be able to check for an existing file and check that file for an ID if the file doesn't exist or doesn't support storing an an ID Sonarr can use for a lookup it would return null.